subsys/nfc/ndef/ch_rec_parser.c

```diff
@@ -167,24 +167,31 @@ static int ac_rec_payload_parse(struct nfc_ndef_bin_payload_desc *payload_desc,
 	uint32_t payload_len = payload_desc->payload_length;
 	const uint32_t buf_size = *result_buf_len;
 	struct net_buf_simple buf;
-	struct nfc_ndef_ch_ac_rec_ref carrier_data_ref;
-	struct nfc_ndef_ch_ac_rec_ref *aux_data_ref;
-	uint8_t aux_data_ref_cnt;
+	struct nfc_ndef_ch_ac_rec *ac_rec;
 
 	/* Initialize memory allocator with buffer memory block and
 	 * its size.
 	 */
 	net_buf_simple_init_with_data(&buf, result_buf, buf_size);
 	net_buf_simple_reset(&buf);
 
+	ac_rec = (struct nfc_ndef_ch_ac_rec *)
+			memory_allocate(&buf, sizeof(*ac_rec));
+	if (!ac_rec) {
+		return -ENOMEM;
+	}
+
+	memset(ac_rec, 0, sizeof(*ac_rec));
```

Comment:
Is it an issue here ? ac-rec is just a placeholder here. I would like to provide some better solution for it but I would like to do it in separated PR

Reply:
This was a fix to a finding from the static code analysis tool. Maybe let's fix the finding in another PR, and let this one go as it fixes the runtime issue which persists since the last release

Reply:
After discussion with @KAGA164 LGTM. There will be changes to the struct as a whole to fix the `const` issue.